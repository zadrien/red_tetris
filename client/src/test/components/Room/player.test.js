import React from 'react'
import { expect } from 'chai'

import { shallow, mount } from 'enzyme'
import ConnectedPlayer, { Players } from '../../../components/Room/Player'

describe('Other player card', () => {
    const setup = (props) => {
        const wrapper = shallow(<Players {...props}/>)

        return {
            props,
            wrapper
        }
    }

    it('should render', () => {
        let props = {
            players: [
                {
                    name: 'testName1',
                    display: [['.', '.', '.', '.', '.', '.', '.', '.', '.', '.']]
                },
                {
                    name: 'testName2',
                    display: [['.', '.', '.', '.', '.', '.', '.', '.', '.', '.']]                },
                {
                    name: 'testName3',
                    display: [['.', '.', '.', '.', '.', '.', '.', '.', '.', '.']]                    
                }
            ]
        }
        const { wrapper } = setup(props)
        expect(wrapper.find('div').children('Card').length).to.be.equal(3)
    })
})