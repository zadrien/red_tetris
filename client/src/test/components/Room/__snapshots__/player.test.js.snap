// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mini board handler should render 1`] = `
<div
  className="d-flex row"
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "marginLeft": "5px",
    }
  }
>
  <div
    className="mini-board"
  >
    <h6>
      testName1
    </h6>
    <div>
      <div
        className="board-row"
      >
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="mini-board"
  >
    <h6>
      testName2
    </h6>
    <div>
      <div
        className="board-row"
      >
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="mini-board"
  >
    <h6>
      testName3
    </h6>
    <div>
      <div
        className="board-row"
      >
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
        <div
          className="mini-board-box"
          style={
            Object {
              "backgroundColor": "Black",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
